SectionedCollectionView/Classes/SectionedCollectionView.swift

```diff
@@ -175,25 +184,43 @@ public class SectionedCollectionView: UIView {
         collectionView.delegate = self
     }
     
-}
-
-extension SectionedCollectionView: UICollectionViewDelegate {
-
-    public func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
-        let selectedItemsCount = self.sections.compactMap({ sections -> [Selectable] in
+    fileprivate func didSelectedItem(_ indexPaths: [IndexPath]) {
+        var selectedItemsCount = self.sections.compactMap({ sections -> [Selectable] in
             return sections.selectedItems()
         }).flatMap({ $0 }).count
-
-        if (self.sections[indexPath.section].items[indexPath.row].selected || self.settings.data.selectedLimit == nil || selectedItemsCount < self.settings.data.selectedLimit!) {
-            self.sections[indexPath.section].items[indexPath.row].selected = !self.sections[indexPath.section].items[indexPath.row].selected
+        
+        let items = indexPaths.map { sections[$0.section].items[$0.row] }
+        let itemsToSelect = items.filter { !$0.selected }.count
+        let itemsToDeselect = items.filter { $0.selected }.count
+        
+        selectedItemsCount += (itemsToSelect - itemsToDeselect)
+        
+        if (self.settings.data.selectedLimit == nil || selectedItemsCount <= self.settings.data.selectedLimit!) {
+            for index in indexPaths {
+                self.sections[index.section].items[index.row].selected = !self.sections[index.section].items[index.row].selected
```

Comment:
let's update this to always select the items programmatically and not depending of its previous state

Reply:
And also let's add a new method to deselect the items in the same way we have the select item